### PR TITLE
Docteam 743 cleanup

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
Clean up non-content items in the docs repo including server docs "guide" pages and contributors guide.

# Reasons
* Some content belongs in the style guide
* Some should be in Confluence
* The scope of what open source contributors can do has changes post-re-platform

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -21,34 +18,21 @@ These contributions come in two forms: issues and pull requests.
...
 
-If you spot anything
-that conflicts with our values,
-opening a GitHub Issue is a great way
-to give us specific feedback.
+If you spot anything that conflicts with our values, opening a GitHub Issue is a great way to give us specific feedback.
 
-To make an issue,
-refer to the [GitHub Issues Workflow](https://github.com/circleci/circleci-docs/wiki/GitHub-Issues-Workflow) wiki page.
+To make an issue, refer to the [GitHub Issues Workflow](https://github.com/circleci/circleci-docs/wiki/GitHub-Issues-Workflow) wiki page.
```
Comment (devinmarieb) [inline docs/CONTRIBUTING.md:23]: Maybe change "make an issue" to "create an issue"? I'd also maybe make this more concise and add it with the above sentence. 

> If you spot anything that conflicts with our values, opening a GitHub Issue is a great way to give us specific feedback. Refer to the [GitHub Issues Workflow](https://github.com/circleci/circleci-docs/wiki/GitHub-Issues-Workflow) wiki page to create an issue.

File: docs/CONTRIBUTING.md [hunk 2]
```diff
@@ -21,34 +18,21 @@ These contributions come in two forms: issues and pull requests.
...
-where you can submit a pull request for your changes.
+For minor changes like typos, click "Suggest an edit to this page", located at the bottom of each document.
+This will take you to the source file on GitHub, where you can submit a pull request for your changes.
 
-For larger edits or new documents,
-[set up a local environment](local-development.md).
-When you are satisfied with your changes,
-create a pull request from your branch
-by following [GitHub's guide](https://help.github.com/articles/creating-a-pull-request-from-a-fork/).
+For larger edits or new documents, [set up a local environment](local-development.md).
```
Comment (devinmarieb) [inline docs/CONTRIBUTING.md]: I thought the local environment no longer existed? I would clarify that someone can fork the repo to work locally, but there is really no "environment" to set up anymore as someone can't run the site locally.

File: docs/CONTRIBUTING.md [hunk 3]
```diff
@@ -48,3 +50,65 @@ Every pull request should have a description that explains *why* the change is b
...
+`page-layout` and the page title are the only required variables. `page-layout` associates an HTML layout with the page, `classic-docs` is used across our docs. `title` will appear at the top of your article.
+
+## Docker Tag List for CircleCI Convenience Images
+
+The Docker tag list for convenience images, located in ./jekyll/_cci2/circleci-images.md, is dynamically updated during a CircleCI build.
+There's usually no need to touch this.
+If you would like to see an updated list generated locally however, you can do so by running `./scripts/pull-docker-image-tags.sh` from the root of this repo.
+Note that you will need the command-line tool [jq](https://stedolan.github.io/jq/) installed.
+
+## Updating `browserlist-stats.json`
```
Comment (devinmarieb) [inline docs/CONTRIBUTING.md:105]: Is this still relevant? This seems like something that my live with docs-platform now.

File: docs/CONTRIBUTING.md [hunk 4]
```diff
@@ -48,3 +50,65 @@ Every pull request should have a description that explains *why* the change is b
...
+:page-liquid:
+:page-description: A short description of the content on the page.
+```
+
+`page-layout` and the page title are the only required variables. `page-layout` associates an HTML layout with the page, `classic-docs` is used across our docs. `title` will appear at the top of your article.
+
+## Docker Tag List for CircleCI Convenience Images
+
+The Docker tag list for convenience images, located in ./jekyll/_cci2/circleci-images.md, is dynamically updated during a CircleCI build.
+There's usually no need to touch this.
```
Comment (devinmarieb) [inline docs/CONTRIBUTING.md:101]:
```suggestion
There is usually no need to touch this.
```

File: docs/CONTRIBUTING.md (hunk 2 above)
Comment (rosieyohannan) [reply docs/CONTRIBUTING.md]: Ah it's not in the final diff

File: docs/CONTRIBUTING.md (hunk 3 above)
Comment (rosieyohannan) [reply docs/CONTRIBUTING.md:105]: Good question. @rsylvian I think we can remove this section from our contributing guide but just want to confirm.

Comment (rsylvian) [reply docs/CONTRIBUTING.md:103]: Correct! :) Thanks for the cleanup!

**Review by devinmarieb — APPROVED** [review]
@rosieyohannan Thanks!

**Pull request merged**